connection.js

```diff
@@ -651,411 +630,140 @@ class Connection {
-                }
-                if (config[minFamilyArray[2]] && !notification.ranked) {
-                    timeNumber = config[minFamilyArray[2]];
-                    timeToString = timespanToDisplayString(config[minFamilyArray[2]]);
-                    rankedUnrankedGamesIsSentence = "unranked games is ";
-                }
+            function minmaxRejectTests(familyArray) {
+                if (config[familyArray[0]] || config[familyArray[1]] || config[familyArray[2]]) {
+                    let argNumber = -1;      // 60
+                    let argToString = "";    // "1 minutes"
+                    let rankedUnranked = ""; // "" or "ranked" or "unranked"
+
+                    if (config[familyArray[0]] && !config[familyArray[1]] && !config[familyArray[2]]) {
+                        argNumber = config[familyArray[0]]; // ex: config.min(main/period)timeblitz
+                        argToString = timespanToDisplayString(config[familyArray[0]]);
+                    } else if (config[familyArray[1]] && notification.ranked) {
+                        argNumber = config[familyArray[1]]; // ex: config.min(main/period)timeblitzranked
+                        argToString = timespanToDisplayString(config[familyArray[1]]);
+                        rankedUnranked = "ranked";
+                    } else if (config[familyArray[2]] && !notification.ranked) {
+                        argNumber = config[familyArray[2]]; // ex: config.min(main/period)blitzunranked
+                        argToString = timespanToDisplayString(config[familyArray[2]]);
+                        rankedUnranked = "unranked";
+                    }
 
-                if ((timecontrolString === "fischer") && ((t.initial_time < timeNumber) || (t.max_time < timeNumber))) {
-                    return UHMAEATForMaintimesFischerRejectProcessing();
-                }
-                if ((timecontrolString === "byoyomi") && t.main_time < timeNumber) {
-                    return UHMAEATForMaintimesByoyomiRejectProcessing();
-                }
-                if ((timecontrolString === "canadian") && t.main_time < timeNumber) {
-                    return UHMAEATForMaintimesCanadianRejectProcessing();
-                }
-                if ((timecontrolString === "absolute") && t.total_time < timeNumber) {
-                    return UHMAEATForMaintimesAbsoluteRejectProcessing();
+                    const timecontrolsSettings = timecontrolsMainPeriodTime(mainPeriodTime, argNumber);
+
+                    let timeCondition = false;
+                    for (let setting of timecontrolsSettings) {
+                        if (familyArray[6]) {            // min
+                            timeCondition = setting[3];  // min condition
+                        } else if (familyArray[7]) {     // max
+                            timeCondition = setting[4];  // max condition
+                        } if (t.time_control === setting[0] && timeCondition) {
```

Comment:
This if should be on the next line by itself. It isn't connected to the prior if statement.